File: libplugin/src/main/java/com/lemon/plugin/inject/InjectPlugin.java
package com.lemon.plugin.inject;

import com.android.build.gradle.AppExtension;
import com.android.build.gradle.LibraryExtension;
import com.android.build.gradle.TestedExtension;

import org.gradle.api.Action;
import org.gradle.api.Plugin;
import org.gradle.api.Project;

import groovy.lang.Closure;
import javassist.ClassPool;

public class InjectPlugin implements Plugin<Project> {

    @Override
    public void apply(Project project) {
        InjectUtils.println("apply plugin:" + project.getName());
        TestedExtension appExtension = project.getExtensions().findByType(AppExtension.class);
        TestedExtension libExtension = project.getExtensions().findByType(LibraryExtension.class);
        if(libExtension != null) {
            InjectUtils.println("type:lib");
            libExtension.registerTransform(new ClassTransform(project, false, libExtension));
        }
        if(appExtension != null) {
            InjectUtils.println("type:app");
            appExtension.registerTransform(new ClassTransform(project, true, appExtension));
        }
    }
}

File: libplugin/src/main/groovy/com/creation/plugin/inject/InjectPlugin.kt
package com.creation.plugin.inject

import com.android.build.gradle.AppExtension
import com.android.build.gradle.LibraryExtension

import org.gradle.api.Plugin
import org.gradle.api.Project

class InjectPlugin : Plugin<Project> {
    override fun apply(project: Project) {
        InjectUtils.println("apply plugin:" + project.name)
        val appExtension = project.extensions.findByType(AppExtension::class.java)
        val libExtension = project.extensions.findByType(LibraryExtension::class.java)
        if (libExtension != null) {
            InjectUtils.println("type:lib")
            libExtension.registerTransform(ClassTransform(project, false, libExtension))
        }
        if (appExtension != null) {
            InjectUtils.println("type:app")
            appExtension.registerTransform(ClassTransform(project, true, appExtension))
        }
    }
}
